#24 Feat : 로그인 - 비밀번호 90일마다 재설정

uri010 opened the pull request:
## 요약
- 비밀번호 90일마다 재설정

## 변경 내용
- 비밀번호를 설정한 날짜를 저장하는 password_date 필드를 user 테이블에 추가합니다.
- password_date 값과 현재 날짜와 90일 이상 차이나면 AuthorizationException(EXCEEDED_PASSWORD_VALIDITY_PERIOD) 예외 처리로 비밀번호를 재설정해야 한다는 메시지를 전송합니다. 

## Issue number/URL
#24 Feat : 로그인 - 비밀번호 90일마다 재설정

uri010 committed:
#24 feat : 비밀번호 90일마다 재설정

비밀번호를 설정한 날짜를 저장하는 password_date 필드를 user 테이블에 추가
password_date 값과 현재 날짜와 90일 이상 차이나면 AuthorizationException(EXCEEDED_PASSWORD_VALIDITY_PERIOD) 예외 처리로 비밀번호 재설정 필수

sunwootest reviewed (commented):
지난번에 피드백한 내용이긴 하지만 User로 충분한 내용을 LoginRes 혹은 Req과 같은 형태로 분리해서 코드 가독성이 떨어지는 것 같습니다.

uri010 commented:
> 지난번에 피드백한 내용이긴 하지만 User로 충분한 내용을 LoginRes 혹은 Req과 같은 형태로 분리해서 코드 가독성이 떨어지는 것 같습니다.

controller에서 Validation 처리를 해주고 싶어서 Res,Req로 나누었습니다. 혹시 LoginReq는 두고 LoginRes를 User로 바꾸는 건 어떤가요?

sunwootest commented:
> > 지난번에 피드백한 내용이긴 하지만 User로 충분한 내용을 LoginRes 혹은 Req과 같은 형태로 분리해서 코드 가독성이 떨어지는 것 같습니다.
> 
> controller에서 Validation 처리를 해주고 싶어서 Res,Req로 나누었습니다. 혹시 LoginReq는 두고 LoginRes를 User로 바꾸는 건 어떤가요?

타당한 근거가 있다면 사용하셔도 됩니다. LoginReq는 필요할 것 같은데 Res가 굳이 필요할지는 모르겠네요 그건 User로 바꿔도 될 것 같습니다 :)

uri010 committed:
#24 fix : LoginRes를 User로 수정 및 UserConstant의 필드를 생성자 바인딩으로 수정

uri010 commented:
## 요약 
- LoginRes를 User로 수정
- UserConstant 필드를 생성자 바인딩으로 수정
## 변경 내용
- LoginRes를 User로 수정해 가독성을 높입니다.
- UserConstant의 필드들을 생성자 바인딩으로 application.properties의 값을 매핑하여 불변성을 유지합니다.

sunwootest commented on src/main/java/study/sharedcalendar/service/LoginService.java:
```diff
@@ -24,22 +22,25 @@ public class LoginService {
+        User user = userMapper.findLoginUser(loginReq);
 
-        if (loginRes.getUserId() == null) {
+        if (user == null || user.getUserId() == null) {
```
userId의 null 여부는 왜 확인하는건가요?

uri010 replied:
userId가 제대로 매핑이 안되는거 같아 임시적으로 처리해두었습니다. 수정하겠습니다.

sunwootest commented on src/main/java/study/sharedcalendar/service/LoginService.java:
```diff
@@ -24,22 +22,25 @@ public class LoginService {
-            loginTryCountCheck(loginRes.getTryCount());
-            userMapper.incrementLoginTryCount(loginRes.getId());
+        if (!encryptionService.isMatch(loginReq.getPassword(), user.getPassword())) {
+            loginTryCountCheck(user.getTryCount());
+            userMapper.incrementLoginTryCount(user.getId());
             throw new AuthorizationException(NO_MATCHING_USER_PASSWORD);
         }
-        if (!loginRes.isActivate()) {
+        if (!user.isActivate()) {
```
if문 사이에 개행이 있으면 가독성이 더 좋아질 거 같습니다 :)

sunwootest commented on src/main/java/study/sharedcalendar/constant/UserConstant.java:
```diff
@@ -1,15 +1,18 @@
+    public UserConstant(int maxLoginTryCount, int maxPasswordValidityPeriod) {
+        this.maxLoginTryCount = maxLoginTryCount;
+        this.maxPasswordValidityPeriod = maxPasswordValidityPeriod;
+    }
```
지난 PR에서 말씀드리긴 했는데 상수값은 getter를 호출하지 말고 static으로 선언한 후 사용하는 것이 더 나아보입니다.

uri010 committed:
#24 fix : UserConstant 상수 static으로 관리 및 코드 정리

UserConstant의 필드들을 static 상수로 관리해 getter를 사용하지 않고 값을 불러옴
if문 사이에 개행을 넣어 가독성을 높임
userId가 null인지 체크하는 부분 삭제

uri010 commented:
## 요약
- UserConstant 상수 static으로 관리
- 코드 정리
## 변경 내용
- UserConstant의 필드들을 static 상수로 관리해 getter를 사용하지 않고 값을 불러옵니다.
- LoginService의 login 메소드에서 if문들 사이에 개행을 넣어 가독성을 높입니다.
- userId가 null인지 체크하는 부분을 삭제합니다.

sunwootest reviewed (commented):
임시 approve 드립니다 :)
